runtime/src/iree/hal/utils/pending_queue_actions.h

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_HAL_UTILS_PENDING_QUEUE_ACTIONS_H_
+#define IREE_HAL_UTILS_PENDING_QUEUE_ACTIONS_H_
+
+#include <stdbool.h>
+
+#include "iree/base/api.h"
+#include "iree/base/internal/arena.h"
+#include "iree/hal/api.h"
+#include "iree/hal/utils/event_pool.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Implementation type for GPU streams.
+// It maps to CUevent for CUDA and hipEvent_t for HIP.
+typedef struct iree_hal_stream_impl_t* iree_hal_stream_impl_t;
+// Implementation type for GPU graph executable.
+// It maps to CUgraphExec for CUDA and hipGraphExec_t for HIP.
+typedef struct iree_hal_graph_executable_impl_t*
+    iree_hal_graph_executable_impl_t;
```

Comment:
Do we have abstractions for "streams" and "graphs" in the base IREE HAL? Should we?

Are there other [GPU] APIs with similar concepts that may want to use this too? (Metal, D3D12, etc.)